### PR TITLE
Add Jabberwocky Keyboard (Attempt 2)

### PR DESCRIPTION
**Pull request by nopunin10did**
## Description
Adds firmware for the new Jabberwocky keyboard.  First PR was #12452, but Travis keeps breaking on that one because it's not as up-to-date with master.  If this PR passes the Travis build, I'll close the other one.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by nopunin10did** [comment]
@drashna I got Travis to stop crapping out on this branch build, and this PR should include all the changes you previously suggested.

File: keyboards/nopunin10did/jabberwocky/jabberwocky.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "jabberwocky.h"
```
Comment (drashna) [inline keyboards/nopunin10did/jabberwocky/jabberwocky.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/nopunin10did/jabberwocky/jabberwocky.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_RH_ANY( \
```
Comment (drashna) [inline keyboards/nopunin10did/jabberwocky/jabberwocky.h:21]:
```suggestion
#define LAYOUT_rh_any( \
```

File: keyboards/nopunin10did/jabberwocky/jabberwocky.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	{ KC_NO, K0105, K0205, KC_NO, K0405, K0505, K0605, K0705, K0805, K0905 }, \
+	{ K0006, K0106, K0206, K0306, K0406, K0506, K0606, K0706, K0806, K0906 }, \
+	{ K0007, K0107, K0207, K0307, K0407, K0507, K0607, K0707, K0807, K0907 }, \
+	{ K0008, K0108, K0208, K0308, K0408, K0508, K0608, K0708, K0808, K0908 }, \
+	{ K0009, K0109, K0209, K0309, K0409, K0509, K0609, K0709, K0809, K0909 }, \
+	{ K0010, K0110, K0210, K0310, K0410, K0510, K0610, K0710, K0810, K0910 }, \
+	{ KC_NO, KC_NO, KC_NO, K0311, K0411, K0511, K0611, K0711, K0811, K0911 }  \
+}
+
+#define LAYOUT_LH_ANY( \
```
Comment (drashna) [inline keyboards/nopunin10did/jabberwocky/jabberwocky.h:44]:
```suggestion
#define LAYOUT_lh_any( \
```

File: keyboards/nopunin10did/jabberwocky/info.json [hunk 4]
```diff
@@ -0,0 +1,127 @@
+{
+    "keyboard_name": "Jabberwocky",
+    "url": "https://github.com/qmk/qmk_firmware/tree/master/keyboards/nopunin10did/railroad",
+    "maintainer": "NoPunIn10Did",
+    "manufacturer": "NoPunIn10Did",
+    "width": 23.5,
+    "height": 7.75,
+    "layouts": {
+        "LAYOUT_RH_ANY": {
```
Comment (drashna) [inline keyboards/nopunin10did/jabberwocky/info.json:9]:
```suggestion
        "LAYOUT_lh_any": {
```

File: keyboards/nopunin10did/jabberwocky/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[0] = LAYOUT_RH_ANY(
```
Comment (drashna) [inline keyboards/nopunin10did/jabberwocky/keymaps/default/keymap.c:21]:
```suggestion
[0] = LAYOUT_rh_any(
```

File: keyboards/nopunin10did/jabberwocky/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+KC_ESC ,KC_F1  ,KC_F2  ,KC_F3  ,KC_F4  ,KC_F5  ,KC_F6  ,                KC_F7  ,KC_F8  ,KC_F9  ,KC_F10 ,KC_F11 ,KC_F12 ,KC_PSCR,KC_SLCK,KC_INS ,KC_DEL ,KC_HOME,KC_END ,KC_PGUP,KC_PGDN,    
+        KC_GRV ,KC_1   ,KC_2   ,KC_3   ,KC_4   ,KC_5   ,                KC_6   ,KC_7   ,KC_8   ,KC_9   ,KC_0   ,KC_MINS,KC_EQL ,KC_BSPC,KC_BSPC,        KC_NLCK,KC_PSLS,KC_PAST,KC_PAUS,    
+        KC_TAB ,KC_Q   ,KC_W   ,KC_E   ,KC_R   ,KC_T   ,                KC_Y   ,KC_U   ,KC_I   ,KC_O   ,KC_P   ,KC_LBRC,KC_RBRC,KC_BSLS,                KC_P7  ,KC_P8  ,KC_P9  ,KC_PMNS,     
+        KC_CAPS,KC_A   ,KC_S   ,KC_D   ,KC_F   ,KC_G   ,                KC_H   ,KC_J   ,KC_K   ,KC_L   ,KC_SCLN,KC_QUOT,KC_ENT ,KC_ENT ,                KC_P4  ,KC_P5  ,KC_P6  ,KC_PPLS,     
+KC_LSFT,KC_LSFT,KC_Z   ,KC_X   ,KC_C   ,KC_V   ,KC_B   ,MO(1),  MO(1),  KC_N   ,KC_M   ,KC_COMM,KC_DOT ,KC_SLSH,KC_RSFT,KC_RSFT,KC_UP  ,                KC_P1  ,KC_P2  ,KC_P3  ,KC_PENT,     
+        KC_LCTL,KC_LGUI,        KC_LALT,KC_MENU,KC_SPC ,                KC_SPC ,        KC_ALGR,        KC_RGUI,KC_RCTL,KC_LEFT,KC_DOWN,KC_RGHT,        KC_P0  ,KC_P0  ,KC_PDOT,KC_PENT),   
+    
+    
+[1] = LAYOUT_RH_ANY(
```
Comment (drashna) [inline keyboards/nopunin10did/jabberwocky/keymaps/default/keymap.c:31]:
```suggestion
[1] = LAYOUT_rh_any(
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/nopunin10did/jabberwocky/info.json (hunk 4 above)
Comment (nopunin10did) [reply keyboards/nopunin10did/jabberwocky/info.json]: Recent commit changed to LAYOUT_rh_any

File: keyboards/nopunin10did/jabberwocky/jabberwocky.c (hunk 1 above)
Comment (nopunin10did) [reply keyboards/nopunin10did/jabberwocky/jabberwocky.c:17]: Added

**Comment by nopunin10did** [comment]
@drashna I think this latest commit has all the further changes you asked for

File: keyboards/nopunin10did/jabberwocky/readme.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+Both the Dexterous and Sinister use the same firmware.  The numpad and the four keys above it are simply transposed from the right to the left side of the board.
+
+For more information, join the [NoPunIn10Key Discord](https://discord.gg/sku2Y6w).
+
+* Keyboard Maintainer: [Alex Ronke](diplomacyvariants.wordpress.com), also known as NoPunIn10Did. 
+* Hardware Supported: Dexterous Jabberwocky and Sinister Jabberwocky (2021 March)
+* Hardware Availability: Currently by request only
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/nopunin10did/jabberwocky/readme.md:17]: Could you add instructions on how to jump to the bootloader, as well?

Comment (nopunin10did) [reply keyboards/nopunin10did/jabberwocky/readme.md:17]: Will do. It's a simple reset switch.

Comment (nopunin10did) [reply keyboards/nopunin10did/jabberwocky/readme.md:23]: Just added.

**Review by drashna — APPROVED** [review]

File: keyboards/nopunin10did/jabberwocky/rules.mk [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (fauxpark) [inline keyboards/nopunin10did/jabberwocky/rules.mk:20]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/nopunin10did/jabberwocky/rules.mk (hunk 8 above)
Comment (nopunin10did) [reply keyboards/nopunin10did/jabberwocky/rules.mk]: Suggested change made

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**